client/zeth/cli/zeth_mix.py

```diff
@@ -73,27 +86,46 @@ def mix(
         raise ClickException("input and output value mismatch")
 
     eth_address = load_eth_address(eth_addr)
-    eth_private_key_data = load_eth_private_key(eth_private_key)
 
     # If instance uses an ERC20 token, tx_value can be 0. Otherwise it should
     # match vin_pub.
     tx_value = EtherValue(0) if mixer_desc.token else vin_pub
 
     # Create the MixParameters object manually so they can be displayed.
     # TODO: support saving the generated MixParameters to be sent later.
-    mix_params, _ = zeth_client.create_mix_parameters_and_signing_key(
-        prover_client,
-        wallet.merkle_tree,
-        zeth_address.ownership_keypair(),
-        eth_address,
-        inputs,
-        outputs,
-        vin_pub,
-        vout_pub)
+    mix_params, signing_keypair = \
+        zeth_client.create_mix_parameters_and_signing_key(
+            prover_client,
+            wallet.merkle_tree,
+            zeth_address.ownership_keypair(),
+            eth_address,
+            inputs,
+            outputs,
+            vin_pub,
+            vout_pub,
+            for_dispatch_call=for_dispatch_call)
 
-    if show_parameters:
-        print(f"mix_params={mix_params.to_json()}")
+    # Dump parameters if requested
+    if dump_parameters:
+        if dump_parameters == '-':
+            print(f"mix_params={mix_params.to_json()}")
```

Comment:
I'm happy to change this, but there are a few things to consider.
I generally put some kind of identifier in front of logging output, to track where it comes from, however I wouldn't really consider this "logging" as such.  It's more like principle output from the command (since the user has explicitly requested it with a flag).  From that point of view, my instinct would be to just output the json (without the `mix_params=` prefix), however this is could be confusing in this case, since the user may be asking for multiple things to be output from a single command (hence I added this "label" prefix).
If it's an issue, we could just not support outputing to stdout.  It might be convenient in somce cases, but it can be easily worked around in a script.
Let me know what you think.

Reply:
I see, thanks for clarifying the intention. Sounds like all the `<function_name>: <message>` logs would benefit from being output with a logging tool then (via `logging.info()/logging.debug()` for instance https://docs.python.org/3/library/logging.html) instead of being dropped on stdout with `print` (and thus mixing with user requested output).

Not a big deal for now, but that would be a good improvement.

Reply:
Yeah.  It needs a bit of thought, but generally that's the approach for those messages with a `<fucntion-name>: ` prefix.  I added a comment to the issue.